src/main/java/alfred/action/Parser.java

```diff
@@ -0,0 +1,246 @@
+                if (descriptions.length == 0) {
+                    throw new MissingDescriptionException();
+                } else {
+                    String[] inputs = input.split("/by ");
+                    if (!isValidTime(inputs[1]) || !isValidDate(inputs[1])) {
+                        throw new InvalidDateException();
+                    } else {
+                        try {
+                            Task newTask = new Deadline(inputs[0], getDate(inputs[1]),
+                                    getTime(inputs[1]));
+                            currentTasks.addTasks(newTask);
+                            input = Ui.sayAdd(user_input, newTask, currentTasks);
+                            currentStorage.appendTaskToFile(newTask);
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }
+            } else if (input.contains(Commands.COMMAND_EVENT)) {
+                String[] descriptions = input.split(Commands.COMMAND_EVENT);
+                if (descriptions.length == 0) {
+                    throw new MissingDescriptionException();
+                } else {
+                    String[] inputs = input.split("/at ");
+                    if (!isValidTime(inputs[1]) || !isValidDate(inputs[1])) {
+                        throw new InvalidDateException();
+                    } else {
+                        try {
+                            Task newTask = new Event(inputs[0], getDate(inputs[1]),
+                                    getTime(inputs[1]));
+                            currentTasks.addTasks(newTask);
+                            input = Ui.sayAdd(user_input, newTask, currentTasks);
+                            currentStorage.appendTaskToFile(newTask);
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }
+            } else if (input.contains(Commands.COMMAND_DELETE)) {
+                String[] descriptions = input.split(Commands.COMMAND_DELETE);
+                if (descriptions.length == 0) {
+                    throw new EmptyInputException();
+                } else {
+                    int taskNumber = findDigit(input);
+                    Task removedTask = currentTasks.getTask(taskNumber);
+                    currentTasks.removeTasks(taskNumber);
+                    input = Ui.sayDelete(user_input, removedTask, currentTasks);
+                    try {
+                        currentStorage.writeTasksToFile(currentTasks);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            } else if (input.equals("")) {
+                throw new EmptyInputException();
+            } else {
+                throw new InvalidInputException();
```

Comment:
Would this be less troublesome to maintain if you can use case/switch statement instead?